client/pdf_generator/__init__.py

```diff
@@ -0,0 +1,114 @@
+else:
+	has_weasyprint = True
+
+class Plugin(getattr(plugins, 'ClientPluginMailerAttachment', plugins.ClientPlugin)):
+	authors = ['Jeremy Schoeneman', 'Erik Daguerre']
+	classifiers = ['Plugin :: Client :: Email :: Attachment']
+	title = 'Generate PDF'
+	description = """
+	Generates a PDF file from an html attachment that process client King Phisher Jinja variables
+	allowing to embed links to your landing page so users that click the link in the PDF can be tracked
+	when they visit.
+	"""
+	homepage = 'https://github.com/securestate/king-phisher-plugins'
+	options = [
+		plugins.ClientOptionPath(
+			'css_stylesheet',
+			'css stylesheet to use for HTML to PDF',
+			display_name='css stylesheet',
+			path_type='file-open'
+		)
+	]
+	req_min_version = '1.8.0'
+	req_packages = {
+		'weasyprint==47': has_weasyprint
+	}
+	version = '2.0'
+	def initialize(self):
+		mailer_tab = self.application.main_tabs['mailer']
+		self.add_menu_item('Tools > Create PDF Preview', self.make_preview)
+		return True
+
+	def make_preview(self, _):
+		input_path = self.application.config['mailer.attachment_file']
+		if not os.path.isfile(input_path) and os.access(input_path, os.R_OK):
+			gui_utilities.show_dialog_error(
+				'PDF Build Error',
+				self.application.get_active_window(),
+				'Attachment path is invalid or is not readable.'
+			)
+			return
+
+		dialog = extras.FileChooserDialog('Save Generated PDF File', self.application.get_active_window())
+		response = dialog.run_quick_save('preview.pdf')
+		dialog.destroy()
+		if response is None:
+			return
+
+		output_path = response['target_path']
+		if not self.process_attachment_file(input_path, output_path):
+			return
+		gui_utilities.show_dialog_info(
+			'PDF Created',
+			self.application.get_active_window(),
+			'Successfully created the PDF file.'
+		)
+
+	def process_attachment_file(self, input_path, output_path, target=None):
+		output_path, _ = os.path.splitext(output_path)
+		output_path += '.pdf'
+		formatted_message = None
+		with codecs.open(input_path, 'r', encoding='utf-8') as file_:
+			msg_template = file_.read()
+
+		try:
+			formatted_message = mailer.render_message_template(msg_template, self.application.config, target)
+		except jinja2.exceptions.TemplateSyntaxError as error:
+			gui_utilities.show_dialog_error(
+				'PDF Build Error',
+				self.application.get_active_window(),
+				"Template syntax error: {error.message} on line {error.lineno}.".format(error=error)
+			)
+			return
+		except jinja2.exceptions.UndefinedError as error:
+			gui_utilities.show_dialog_error(
+				'PDF Build Error',
+				self.application.get_active_window(),
+				"Template undefined error: {error.message}.".format(error=error)
+			)
+			return
+		except TypeError as error:
+			gui_utilities.show_dialog_error(
+				'PDF Build Error',
+				self.application.get_active_window(),
+				"Template type error: {0}.".format(error.args[0])
+			)
+			return
+
+		if os.path.isfile(self.config.get('css_stylesheet')):
```

Comment:
If `css_stylesheet` is None which it looks like it could be based on the fact that you're using `.get`, then `os.path.isfile` will raise a `TypeError`.

If `css_stylesheet` can't be None and will only ever be an empty string, then you shouldn't use `.get`. I'd suggest pulling the path into a variable, stripping white space, then evaluating that you have a non-empty string before checking `os.path.isfile`. Should also add a `os.access` check in there too for good measure.